config/package_resource.go

```diff
@@ -101,8 +116,14 @@ func (p *packageResouce) validate(ctx context.Context) (*ManagedResources, error
 		if !packages.DpkgExists {
 			return nil, fmt.Errorf("cannot manage Deb package because dpkg does not exist on the system")
 		}
+		if p.GetDesiredState() != agentendpointpb.ApplyConfigTask_OSPolicy_Resource_PackageResource_INSTALLED {
```

Comment:
Don't we support other desired states(example: REMOVED) for Deb and other package resources that are checking only for Installed state here? 

Do we plan to add support for other desired states in the future PRs?

Reply:
Currently I don't plan on supporting anything but INSTALLED as it really doesnt make much sense, we should also add this check in the API. If you want to remove a deb package use the apt functionality, same with rpm and yum/zypper.

Reply:
Got it. Can you file a bug to update our API documentation and also to add this validation on the service side?

Looking through the cases here, I see that only APT and Googet  allows for REMOVED state. Checking to see if that is what you see as the final support for these package types or if you want to add REMOVED support for more package types in the future. If it is the latter, adding TODOs might help to differentiate the cases.

Also, I don't see a case statement for RPM.

Reply:
Created a bug to track this on the API side. apt, googet, yum and zypper all allow install and remove. deb, rpm and msi only allow install, those three should have a simple if check in validation.